### PR TITLE
Update test_postgresql_manager.py

### PR DESCRIPTION
**Pull request by francescoingv**
# Overview
Add test to check issue #1750 is fixed

# Related Issue / discussion
#1750 

# Additional information
The PR will break test suite until the issue #1750 is not solved.

# Dependency policy (RFC2)

- [X] I have ensured that this PR meets [RFC2](https://pygeoapi.io/development/rfc/2) requirements

# Updates to public demo

- [ ] I have ensured that breaking changes to the [pygeoapi master demo server](https://github.com/geopython/demo.pygeoapi.io) have been addressed
  - [ ] https://github.com/geopython/demo.pygeoapi.io/blob/master/services/pygeoapi_master/local.config.yml

# Contributions and licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [X] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution
- [X] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

File: tests/test_postgresql_manager.py [hunk 1]
```diff
@@ -103,6 +103,15 @@ def _create_delete_request(job_id, locales):
     return APIRequest.with_data(req, locales)
 
 
+def test_api_connection_rfc3986(config, openapi):
+    connection = config['server']['manager']['connection']
+    connectionString = (
```
Comment (tomkralidis) [inline tests/test_postgresql_manager.py:108]: `connectionString` -> `connection_string`

**Review by tomkralidis — APPROVED** [review]

**Review by tomkralidis — CHANGES_REQUESTED** [review]
minor variable name change requested.

**Comment by tomkralidis** [comment]
@francescoingv can you add this update to this PR:

```diff
diff --git a/pygeoapi/process/manager/postgresql.py b/pygeoapi/process/manager/postgresql.py
index 8215b12..fcbe871 100644
--- a/pygeoapi/process/manager/postgresql.py
+++ b/pygeoapi/process/manager/postgresql.py
@@ -46,6 +46,7 @@ from pathlib import Path
 from typing import Any, Tuple
 
 from sqlalchemy import insert, update, delete
+from sqlalchemy.engine import make_url
 from sqlalchemy.orm import Session
 
 from pygeoapi.process.base import (
@@ -88,7 +89,13 @@ class PostgreSQLManager(BaseManager):
         try:
             LOGGER.debug('Connecting to database')
             if isinstance(self.connection, str):
-                self._engine = get_engine(self.connection)
+                _url = make_url(self.connection)
+                self._engine = get_engine(
+                    _url.host,
+                    _url.port,
+                    _url.database,
+                    _url.username,
+                    _url.password)
             else:
                 self._engine = get_engine(**self.connection)
         except Exception as err:
```

**Comment by francescoingv** [comment]
@tomkralidis I made the requested update (https://github.com/geopython/pygeoapi/pull/1751#issuecomment-2246283002)
but the test still fails with the following error: 
```
ERROR    pygeoapi.process.manager.postgresql:postgresql.py:116 Table model fetch failed: Could not connect to ***localhost:5432/test (password hidden).
=========================== short test summary info ============================
FAILED tests/test_postgresql_manager.py::test_api_connection_rfc3986 - pygeoapi.process.base.ProcessorGenericError: Table model fetch failed
```

while in my environment I have the following error:
```
ERROR    pygeoapi.process.manager.postgresql:postgresql.py:116 Table model fetch failed: Table 'jobs' not found in schema 'p' on postgresql+psycopg2://postgres:***@localhost:5432/test.
=========================== short test summary info ============================
FAILED tests/test_postgresql_manager.py::test_api_connection_rfc3986 - pygeoa...

```

**Comment by tomkralidis** [comment]
> @tomkralidis I made the requested update ([#1751 (comment)](https://github.com/geopython/pygeoapi/pull/1751#issuecomment-2246283002)) but the test still fails with the following error:
> 
> ```
> ERROR    pygeoapi.process.manager.postgresql:postgresql.py:116 Table model fetch failed: Table 'jobs' not found in schema 'p' on ***localhost:5432/test.
> =========================== short test summary info ============================
> FAILED tests/test_postgresql_manager.py::test_api_connection_rfc3986 - pygeoapi.process.base.ProcessorGenericError: Table model fetch failed
> ```

@francescoingv looks like tests are now passing.

File: requirements-manager.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
+geoalchemy2
```
Comment (tomkralidis) [inline requirements-manager.txt:1]: Do we need this dependency?  Can we remove?

**Review by tomkralidis — CHANGES_REQUESTED** [review]

**Comment by francescoingv** [comment]
@tomkralidis 
> @francescoingv looks like tests are now passing.

Yes, I added a couple of small commits after the above comment

File: requirements-manager.txt (hunk 2 above)
Comment (tomkralidis) [reply requirements-manager.txt:1]: Can you try removing so we can test what CI says (now that we have fully passing CI)?

Comment (tomkralidis) [reply requirements-manager.txt]: CI passes as a result.  Merging.  Thanks!

Comment (francescoingv) [reply requirements-manager.txt]: @tomkralidis 
I removed the dependency, and it passed the CI.
However it depends on which requirements are installed for each test suite:
`requirements-provider.txt` also includes `psycopg2`
Should I remove also that dependency?

I think `requirements-manager.txt` should include **all and only** the requirements that need to be installed for the manager in addition to `requirements.txt`.
What is included into `requirements-provider.txt` is a different matter.

Or should we rather have a `requirements-postgresql.txt`, or `requirements-sqlalchemy.txt`?

**Pull request merged**